src/main/java/cms/gov/madie/measure/config/UpdateEndorsementsOrgChangeUnit.java

```diff
@@ -0,0 +1,78 @@
+package cms.gov.madie.measure.config;
+
+import cms.gov.madie.measure.repositories.EndorsementRepository;
+import cms.gov.madie.measure.repositories.MeasureRepository;
+import gov.cms.madie.models.common.EndorserOrganization;
+import gov.cms.madie.models.measure.Endorsement;
+import gov.cms.madie.models.measure.Measure;
+import gov.cms.madie.models.measure.MeasureMetaData;
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+@Slf4j
+ @ChangeUnit(id = "endorsements_org_update", order = "2", author = "madie_dev")
+public class UpdateEndorsementsOrgChangeUnit {
+
+      @Execution
+      public void updateEndorsements(
+          MeasureRepository measureRepository, EndorsementRepository endorsementRepository)
+          throws Exception {
+        log.info("Entering updateEndorsements()");
+
+        List<EndorserOrganization> allEndorsers = endorsementRepository.findAll();
+        if (allEndorsers == null || allEndorsers.isEmpty()) {
+          log.error(
+              "No endorsements found! Exiting updateEndorsements to "
+                  + "prevent from destroying all measures!");
+          return;
+        }
+        List<Measure> measureList = measureRepository.findAll();
+        measureList.stream()
+            .forEach(
+                measure -> {
+                  MeasureMetaData measureMetaData = measure.getMeasureMetaData();
+                  boolean updateRequired = false;
+                  List<Endorsement> endorsements = measureMetaData.getEndorsements();
+                  if (endorsements != null && !measureMetaData.getEndorsements().isEmpty()) {
+                    updateRequired = true;
+                    List<Endorsement> updatedOrgs =
+                        endorsements.stream()
+                            .map(
+                                endorsement -> {
+                                  if (!StringUtils.isEmpty(endorsement.getEndorser()) &&
+     StringUtils.isEmpty(endorsement.getEndorsementId())) {
```

Comment:
Can we Trim() before checking if the field is Empty ? Yesterday I have seen few production measures which has a value of "   " idk if that is mongoCloud displaying it.

Reply:
That's a good point. I switched it to isBlank() which will take care of the trim for us without having to worry about the possible null values.